File: javascript/tests/longest_common_prefix_test.js
import {assertEquals} from "@std/assert";
import {longestCommonPrefix} from "@/longest_common_prefix.js";

Deno.test(function exampleCases() {
    const case1 = ["flower", "flow", "flight"];
    const case2 = ["dog", "racecar", "car"];

    assertEquals(longestCommonPrefix(case1), "fl");
    assertEquals(longestCommonPrefix(case2), "");
});

Deno.test(function extraCases() {
    assertEquals(longestCommonPrefix(["foo"]), "foo");
    assertEquals(longestCommonPrefix([""]), "");
    assertEquals(longestCommonPrefix(["a", ""]), "");
});

File: java/src/test/java/com/jamesansley/LongestCommonPrefixTest.java
package com.jamesansley;

import org.junit.jupiter.api.Test;

import static com.jamesansley.Utils.arrayOf;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class LongestCommonPrefixTest {
    @Test
    public void exampleCases() {
        LongestCommonPrefix solution = new LongestCommonPrefix();
        String[] case1 = arrayOf("flower", "flow", "flight");
        String[] case2 = arrayOf("dog", "racecar", "car");

        assertEquals("fl", solution.longestCommonPrefix(case1));
        assertEquals("", solution.longestCommonPrefix(case2));
    }

    @Test
    public void extraCases() {
        LongestCommonPrefix solution = new LongestCommonPrefix();

        assertEquals("foo", solution.longestCommonPrefix(arrayOf("foo")));
        assertEquals("", solution.longestCommonPrefix(arrayOf("")));
        assertEquals("", solution.longestCommonPrefix(arrayOf("a", "")));
    }
}
